client/src/components/Query/Statement.jsx

```diff
@@ -331,20 +334,55 @@ class Statement extends React.Component {
   }
 
   handleRemoveChecked() {
+
     if (this.isRemovable()) {
       const { checkedQueries, draft } = this.state;
-      this.commit(draft);
-      const keysToRemove = checkedQueries.reduce((accumulator, key) => [...accumulator, { key }], []);
-      pullAllBy(draft, keysToRemove, 'key');
-      this.setState({
-        draft,
-        checkedQueries: [],
-        queriesChecksAreIndeterminate: false,
-        queriesAreAllChecked: false,
-      });
+
+      let keys = []
+      for(let d of draft){
+           d.instructions.map(i => {if(i.type === "subquery"){
```

Comment:
@claudia1296 Would it be possible to use [some](https://lodash.com/docs/4.17.15#some) instead?

Reply:
re: pullAllBy, not some. Apologies.